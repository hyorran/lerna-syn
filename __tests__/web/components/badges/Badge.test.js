/* eslint-disable no-undef */
import React from 'react'
import Badge from '@syntesis/c-badges'
import { mount } from 'enzyme'


describe('<Badge />', () => {
  it('rendered', () => {
    expect(Badge).toBeDefined()
    const wrapper = mount((
      <Badge
        content={ 10 }
      >
        <div className="unique" />
      </Badge>
    ))
    expect(wrapper).toMatchSnapshot()
  })
  it('rendered only children', () => {
    const wrapper = mount((
      <Badge invisible content={ 10 }>
        <div className="unique" />
      </Badge>
    ))
    expect(wrapper.contains(<div className="unique" />)).toEqual(true)
  })
})
