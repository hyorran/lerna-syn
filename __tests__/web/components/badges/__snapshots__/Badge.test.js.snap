// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> rendered 1`] = `
<WithStyles(Badge)
  content={10}
>
  <Badge
    classes={
      Object {
        "margin": "Badge-margin-1",
      }
    }
    color="primary"
    content={10}
    invisible={false}
  >
    <WithStyles(Badge)
      badgeContent={10}
      className="Badge-margin-1"
      color="primary"
    >
      <Badge
        badgeContent={10}
        className="Badge-margin-1"
        classes={
          Object {
            "badge": "MuiBadge-badge-3",
            "colorError": "MuiBadge-colorError-6",
            "colorPrimary": "MuiBadge-colorPrimary-4",
            "colorSecondary": "MuiBadge-colorSecondary-5",
            "dot": "MuiBadge-dot-8",
            "invisible": "MuiBadge-invisible-7",
            "root": "MuiBadge-root-2",
          }
        }
        color="primary"
        component="span"
        max={99}
        showZero={false}
        variant="standard"
      >
        <span
          className="MuiBadge-root-2 Badge-margin-1"
        >
          <div
            className="unique"
          />
          <span
            className="MuiBadge-badge-3 MuiBadge-colorPrimary-4"
          >
            10
          </span>
        </span>
      </Badge>
    </WithStyles(Badge)>
  </Badge>
</WithStyles(Badge)>
`;
